flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartitionReader.java

```diff
@@ -89,9 +114,30 @@ public BufferAndBacklog getNextBuffer() throws IOException {
 		return BufferAndBacklog.fromBufferAndLookahead(current, nextBuffer, dataBufferBacklog);
 	}
 
+	/**
+	 * This method is actually only meaningful for the {@link BoundedBlockingSubpartitionType#FILE}. For the
+	 * other types the next buffer is always not null while {@link ResultSubpartitionView#getNextBuffer()},
```

Comment:
```
For the other types the {@link #nextBuffer} can not be ever set to null, so it is (...)
``` 
also if that's true, I think we would need a checkState whenever we set `nextBuffer` to `null`?

Reply:
Do you mean adding `checkState` in `BoundedBlockingSubpartitionReader#getNextBuffer` while next buffer is null? If so, I think it might be not necessary adding it in `getNextBuffer` which is consistent for all the reader types. We could add the check in following `notifyDataAvailable` as you suggested if necessary, because only this method should be only used for file type.